v2/tools/generator/internal/codegen/pipeline/apply_cross_resource_references_from_config.go

```diff
@@ -0,0 +1,237 @@
+	"k8s.io/klog/v2"
+
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/astmodel"
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/config"
+)
+
+type ARMIDPropertyClassification string
+
+// TODO: I don't like the names I am using here
+const (
+	ARMIDPropertyClassificationUnset       = ARMIDPropertyClassification("unset")
+	ARMIDPropertyClassificationSet         = ARMIDPropertyClassification("set")
+	ARMIDPropertyClassificationUnspecified = ARMIDPropertyClassification("unspecified")
+)
+
+// ApplyCrossResourceReferencesFromConfigStageID is the unique identifier for this pipeline stage
+const ApplyCrossResourceReferencesFromConfigStageID = "applyCrossResourceReferencesFromConfig"
+
+// ApplyCrossResourceReferencesFromConfig replaces cross resource references from the configuration with astmodel.ARMID.
+func ApplyCrossResourceReferencesFromConfig(configuration *config.Configuration) *Stage {
+	return NewStage(
+		ApplyCrossResourceReferencesFromConfigStageID,
+		"Replace cross-resource references in the config with astmodel.ARMID",
+		func(ctx context.Context, state *State) (*State, error) {
+			typesWithARMIDs := make(astmodel.TypeDefinitionSet)
+
+			var crossResourceReferenceErrs []error
+
+			isCrossResourceReference := func(typeName astmodel.TypeName, prop *astmodel.PropertyDefinition) ARMIDPropertyClassification {
+				// First check if we know that this property is an ARMID already
+				isReference, err := configuration.ARMReference(typeName, prop.PropertyName())
+				isSwaggerARMID := isTypeARMID(prop.PropertyType())
+
+				// If we've got a Swagger ARM ID entry AND an entry in our config, that might be a problem
+				if isSwaggerARMID && err == nil {
+					if !isReference {
+						// We allow overriding the ARM ID status of a property to false in our config
+						return ARMIDPropertyClassificationUnset
+					} else {
+						// Swagger has marked this field as a reference, and we also have it marked in our
+						// config. Record an error saying that the config entry is no longer needed
+						crossResourceReferenceErrs = append(
+							crossResourceReferenceErrs,
+							errors.Errorf("%s.%s marked as ARM reference, but value is not needed because Swagger already says it is an ARM reference",
+								typeName.String(),
+								prop.PropertyName().String()))
+					}
+				}
+
+				if DoesPropertyLookLikeARMReference(prop) && err != nil {
+					if config.IsNotConfiguredError(err) {
+						// This is an error for now to ensure that we don't accidentally miss adding references.
+						// If/when we move to using an upstream marker for cross resource refs, we can remove this and just
+						// trust the Swagger.
+						crossResourceReferenceErrs = append(
+							crossResourceReferenceErrs,
+							errors.Wrapf(
+								err,
+								"%s.%s looks like a resource reference but was not labelled as one; You may need to add it to the 'objectModelConfiguration' section of the config file",
```

Comment:
```suggestion
								"%s.%s looks like a resource reference but was not labelled as one. You may need to add it to the 'objectModelConfiguration' section of the config file",
```

Reply:
I don't think this change is correct. The second part is a subordinate clause that requires the first part for motivation, it's not an independent sentence.

Reply:
Leaving this as is based on @theunrepentantgeek's comment